src/coordinated_workers/coordinator.py

```diff
@@ -469,19 +488,154 @@ def _reconcile(self):
+                ),
+                proxy_worker_telemetry_port=proxy_worker_telemetry_port,
+                tls_available=self.tls_available,
+                tracing_target_type=_type,
+                protocols=[receiver.protocol.name for receiver in receivers],  # type: ignore
+            )
+
+        return {receiver.protocol.name: receiver.url for receiver in receivers}  # type: ignore
+
     @property
     def _charm_tracing_receivers_urls(self) -> Dict[str, str]:
-        """Returns the charm tracing enabled receivers with their corresponding endpoints."""
-        endpoints = self.charm_tracing.get_all_endpoints()
-        receivers = endpoints.receivers if endpoints else ()
-        return {receiver.protocol.name: receiver.url for receiver in receivers}
+        """Returns the charm tracing enabled receivers per tracing protocol with their corresponding endpoints to be published into the cluster data.
+
+        When worker telemetry proxy is enabled and when not self tracing, returns the proxy url for sending traces.
+
+        Returns:
+            A dictionary of tracing protocols and the respective tracing receiver urls.
+            {
+                "otlp_http": "http://tempo.tempo.svc.cluster.local:4318",
+            }
+
+            Or when worker telemetry proxy is enabled,
+            {
+                "otlp_http": "http://tempo-0.tempo-endpoints.tempo.svc.cluster.local:3300/proxy/charm-tracing/otlp_http/",
+            }
+        """
+        return self._tracing_receivers_urls(self.charm_tracing, "charm-tracing")
 
     @property
     def _workload_tracing_receivers_urls(self) -> Dict[str, str]:
-        """Returns the workload tracing enabled receivers with their corresponding endpoints."""
-        endpoints = self.workload_tracing.get_all_endpoints()
-        receivers = endpoints.receivers if endpoints else ()
-        return {receiver.protocol.name: receiver.url for receiver in receivers}
+        """Returns the workload tracing enabled receivers with their corresponding endpoints to be published into the cluster data.
+
+        When worker telemetry proxy is enabled and when not self tracing, returns the proxy url for sending traces.
+
+        Returns:
+            A dictionary of tracing protocols and the respective tracing receiver urls.
+            {
+                "otlp_http": "http://tempo.tempo.svc.cluster.local:4318",
+            }
+
+            Or when worker telemetry proxy is enabled,
+            {
+                "otlp_http": "http://tempo-0.tempo-endpoints.tempo.svc.cluster.local:3300/proxy/workload-tracing/otlp_http/",
+            }
+        """
+        return self._tracing_receivers_urls(self.workload_tracing, "workload-tracing")
+
+    @property
+    def remote_write_endpoints(self):
+        """Returns the remote write endpoints based on if its available and if proxying telemetry is enabled to be published into the cluster data.
```

Comment:
```suggestion
        """Return the remote write endpoints for prometheus metrics.
        
        Return None if there is no remote_write relation.
        If telemetry proxying is enabled, the endpoints will point to the coordinator's nginx, which maps them to the upstream.
```

General advice (please apply everywhere):
 - use active modus: 'do' instead of 'does'
 - avoid telling what this method is going to be used for: this method shouldn't care about that.
 - avoid complex sentences. If you need to explain further, split into a new sentence.